internal/events/eventhandlers/payment_to_submitter_event_handler.go

```diff
@@ -0,0 +1,82 @@
+package eventhandlers
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/stellar/go/support/log"
+	"github.com/stellar/stellar-disbursement-platform-backend/db"
+	"github.com/stellar/stellar-disbursement-platform-backend/internal/crashtracker"
+	"github.com/stellar/stellar-disbursement-platform-backend/internal/data"
+	"github.com/stellar/stellar-disbursement-platform-backend/internal/events"
+	"github.com/stellar/stellar-disbursement-platform-backend/internal/events/schemas"
+	"github.com/stellar/stellar-disbursement-platform-backend/internal/services"
+	"github.com/stellar/stellar-disbursement-platform-backend/internal/utils"
+	"github.com/stellar/stellar-disbursement-platform-backend/stellar-multitenant/pkg/router"
+	"github.com/stellar/stellar-disbursement-platform-backend/stellar-multitenant/pkg/tenant"
+)
+
+type PaymentToSubmitterEventHandlerOptions struct {
+	DBConnectionPool    db.DBConnectionPool
+	TSSDBConnectionPool db.DBConnectionPool
+	CrashTrackerClient  crashtracker.CrashTrackerClient
+}
+
+type PaymentToSubmitterEventHandler struct {
+	tenantManager      tenant.ManagerInterface
+	crashTrackerClient crashtracker.CrashTrackerClient
+	service            services.PaymentToSubmitterServiceInterface
+}
+
+var _ events.EventHandler = new(PaymentToSubmitterEventHandler)
+
+func NewPaymentToSubmitterEventHandler(options PaymentToSubmitterEventHandlerOptions) *PaymentToSubmitterEventHandler {
+	tm := tenant.NewManager(tenant.WithDatabase(options.DBConnectionPool))
+	tr := router.NewMultiTenantDataSourceRouter(tm)
+	mtnDBConnectionPool, err := db.NewConnectionPoolWithRouter(tr)
+	if err != nil {
+		log.Fatalf("error getting tenant DB Connection Pool: %s", err.Error())
+	}
+
+	models, err := data.NewModels(mtnDBConnectionPool)
+	if err != nil {
+		log.Fatalf("error getting models: %s", err.Error())
+	}
+
+	s := services.NewPaymentToSubmitterService(models, options.TSSDBConnectionPool)
+
+	return &PaymentToSubmitterEventHandler{
+		tenantManager:      tm,
+		service:            s,
+		crashTrackerClient: options.CrashTrackerClient,
+	}
+}
+
+func (h *PaymentToSubmitterEventHandler) Name() string {
+	return "PaymentToSubmitterEventHandler"
+}
+
+func (h *PaymentToSubmitterEventHandler) CanHandleMessage(ctx context.Context, message *events.Message) bool {
+	return message.Topic == events.PaymentCompletedTopic
+}
```

Comment:
🚨 @CaioTeixeira95: I believe this is the wrong topic. Shouldn't it be PaymentReadyToPayTopic?

Reply:
Thank you so much for this!